blog/_posts/2022-03-15-pulumi.md

```diff
@@ -0,0 +1,275 @@
+            spec: { containers: [{ name: "pulumi-helloworld", image: image }] }
+        }
+    }
+});
+exports.name = deployment.metadata.name;
+```
+
+Now, we use Tilt to build the image, live update the running Pod, and connect a port-forward
+from `localhost:8000` to the container deployed with Pulumi.
+
+Tilt's `pulumi` extension invokes `pulumi up` to deploy to the cluster.
+
+```python
+load('ext://pulumi', 'pulumi_resource')
+
+docker_build(
+  'pulumi-helloworld-image',
+  './helloworld',
+  live_update=[
+    sync('./helloworld', '/app'),
+  ])
+
+pulumi_resource(
+  'helloworld',
+  stack='dev',
+  dir='./',
+  deps=['./index.js'],
+  image_deps=['pulumi-helloworld-image'],
+  image_configs=['image'],
+  labels=['helloworld'],
+  port_forwards=['8000:8000'])
+```
+
+The `pulumi_resource` function has arguments to define:
+
+- The name of your resource in the Tilt UI.
+
+- The name of the stack in Pulumi (this is usually implicit when you use the Pulumi CLI).
+
+- What file changes should trigger a re-deploy.
+
+- How to inject the image into your script.
+
+- Any other live updates or port forwards to attach!
+
+A Pulumi Tiltfile is a good way to set up a fast feedback loop so you can
+make small changes -- a little bit at a time -- and verify that they work.
+
+To try this example yourself, check out the [Tilt Pulumi extension
+docs](https://github.com/tilt-dev/tilt-extensions/blob/master/pulumi/README.md)
+or the accompanying
+[example](https://github.com/tilt-dev/tilt-extensions/blob/master/pulumi/test/Tiltfile).
+
+But how do we make sure it all works?
+
+## Verifying Your Pulumi Scripts in CI
+
+The Pulumi engine is built so that you can write unit tests
+against a "fake" backend. The [Pulumi docs](https://www.pulumi.com/docs/guides/testing/)
+have more detail on how to use Pulumi's testing libraries.
+ 
+But we can also use Tilt's blackbox testing to test our Pulumi script against a
+real, one-time-use cluster with `tilt ci`. This is how the Tilt team tests our
+own Pulumi extension!
+
+
+```bash
+#!/bin/bash
+
+cd "$(dirname "$0")"
+set -ex
+
+# install pulumi deps
+yarn install
+
+tilt ci
+```
+
+The `tilt ci` command:
+
+- Builds all the images in the tiltfile.
+
+- Runs `pulumi up` to deploy the images to Kubernetes.
+
+- Tracks the rollout of pods.
+
+- Exits when all the pods become ready.
+
+For more on how to set up a one-time-use cluster for testing your
+infrastructure, see the [Tilt CI Guide](https://docs.tilt.dev/ci.html).
+
+## Future Work
+
+We hope this guide helped you to understand how Pulumi might fit into
+your infrastucture stack, and how to use it for local Kubernetes development.
+
+The Pulumi extension uses a much more full-featured plugin API that lets Tilt use
+any arbitrary Bash script for Kubernetes deployments. To learn more how to adapt Tilt's dev
+environment to other similar tools, check out [Milas' post on
+`k8s_custom_deploy`](https://blog.tilt.dev/2021/12/03/k8s-custom-deploy.html).
```

Comment:
Let's also add a sentence here to encourage people to contribute to this or other Tilt extensions!